brainscore_vision/model_helpers/brain_transformation/behavior.py

```diff
@@ -215,3 +218,87 @@ def labels_to_indices(self, labels):
                 indices.append(label2index[label])
             index2label = OrderedDict((index, label) for label, index in label2index.items())
             return indices, index2label
+
+
+class OddOneOut(BrainModel):
+    def __init__(self, identifier: str, activations_model, layer: Union[str, List[str]]):
+        """
+        :param identifier: a string to identify the model
+        :param activations_model: the model from which to retrieve representations for stimuli
+        :param layer: the single behavioral readout layer or a list of layers to read out of.
+        """
+        self._identifier = identifier
+        self.activations_model = activations_model
+        self.readout = make_list(layer)
+        self.current_task = BrainModel.Task.odd_one_out
+        self.similarity_measure = 'dot'
+
+    @property
+    def identifier(self):
+        return self._identifier
+
+    def start_task(self, task: BrainModel.Task):
+        assert task == BrainModel.Task.odd_one_out
+        self.current_task = task
+
+    def look_at(self, triplets, number_of_trials=1):
+        # 1. get unique stimuli
+        stimuli = None  # TODO Determine unique stimuli from triplets
+        
+        # 2. get activations for stimuli
+        features = self.activations_model(stimuli, layers=self.readout)
+        features = features.transpose('presentation', 'neuroid')
+
+        # 3. calculate similarity matrix
+        similarity_matrix = self.calculate_similarity_matrix(features.values)
+        
+        # 4. calculate choices
+        choices = self.calculate_choices(similarity_matrix, triplets)
+
+        # 5. package and return choices
+        # TODO return as DataAssembly
+        
+        #stimulus_ids = triplets['stimulus_id']
+        #choices = BehavioralAssembly(choice_predictions, coords={
+        #    'triplet_index': ('presentation', [i for i in range(0, len(stimulus_ids), 3)]),
+        #    'triplet_stimulus_id0': ('presentation', [{stimulus_ids[i]} for i in range(0, len(stimulus_ids), 3)]),
+        #    'triplet_stimulus_id1': ('presentation', [{stimulus_ids[i+1]} for i in range(0, len(stimulus_ids), 3)]),
+        #    'triplet_stimulus_id2': ('presentation', [{stimulus_ids[i+2]} for i in range(0, len(stimulus_ids), 3)]),
+        #    'stimulus_id': ('presentation', [f"{stimulus_ids[i]}__{stimulus_ids[i+1]}__{stimulus_ids[i+2]}" for i in range(0, len(stimulus_ids), 3)])
+        #    }, dims=['presentation'])
```

Comment:
This should probably work but as I said above I am not sure how we pass the triplets.